ros_buildfarm/sourcerpm_job.py

```diff
@@ -55,6 +55,22 @@ def build_sourcerpm(
         '--depth', '1', '--no-single-branch',
         release_repository_url, os_pkg_name]
 
+    # HACK FOR RHBZ#1960791
+    # RHEL 8.4 shipped a cmake package which omitted the 'cmake3' virtual
+    # package. This hack replaces dependencies on 'cmake3' with 'cmake>3',
+    # which will work on RHEL 8, but not RHEL 7.
+    if os_name == 'rhel' and os_code_name == '8':
+        fixup_cmd = [
+            'sed', '-i',
+            r's/^\\\(\\\(Build\\\)\\?Requires:\ \*\\\)cmake3$/\\1cmake\>3/g',
```

Comment:
A few questions about the pattern.

* Does this need the `/g` global flag? As far as I understand from the format there is only one dependency per line and the pattern is already tested once per line

* Unless there is truly a situation where `BuildRequires:cmake3` (no space) is a valid line the interior zero-or-more (`*`) should be replaced with a one-or-more `+` metacharacter.

* Do we need to match `(Build)?` at all or can we match just on `Requires:` to keep the pattern simple? since there's vertical alignment being applied we do have to capture at least the spaces so I suppose `^(Build)?Requires: +)cmake3$` makes as much sense as `Requires:( +)cmake3$`

* Are the backslashes necessary? I understand that there's some nested escaping happening between Python and the shell but the command line ` printf 'BuildRequires:          cmake3\Requires:    foobar\nBuildRequires:   science\n' | sed -E 's/Requires(: +)cmake3/Requires\1cmake>3/'`

Reply:
> Does this need the /g global flag?
Nope, dropped in 68e585c
> Unless there is truly a situation where `BuildRequires:cmake3` (no space) is a valid line
This is true. Spaces are not required, though the templates in Bloom always include some. It's valid syntax, so I'd rather support the no-space case.
> Do we need to match (Build)? at all or can we match just on Requires: to keep the pattern simple?
Maybe I'm being overly pedantic, but I'd refer to match the entire line perfectly. I don't feel strongly in either case, but taking the more conservative pattern seems safer.
> Are the backslashes necessary?
There seems to be a big difference in how parentheses are interpreted between the case where `-E` is present and when absent. It turns out that they don't need to be escaped when using `-E`, so that cleaned up the pattern quite a lot. The remaining escapes are for shell characters - note that I'm using a Python raw string to avoid additional escaping. Also note the lack of quoting around the pattern to avoid nested quoting in the command to deal with the `sh -c "foo"`.

Reply:
Got it. Thanks for the rundown.